### DISCUSSION TITLE
Prebuilt Realm Db

### DISCUSSION THREAD

**Discussion by usefulBeeing** [realm/realm-dotnet#3666] [Q&A] [answered]
Just came across realm-dotnet, as an alternative to _sqlite_, and wondering if it might fit my needs. I have a simple scenario where I have a couple of tables that will rarely get changed, with a couple of hundred records at most.  In a _sqlite_ scenario,  I would usually create the database and tables using a tool like _sqlitebrowser_ for example, then ship the prebuilt database, tables and records with the application.

I can't wrap my head around Realm as a local storage solution yet but curious if it could handle a prebuilt objects scenario like this one, and how?

Thank you.

**Comment by peppy** [comment]
In our experience this should work just fine!

**Comment by uixiu** [comment, marked as answer]
My opinions:

Realm database is a better alternative to SQLite even for medium sized database.

PROS:

You get a out of the box

- A good enough Database explorer, with html links to sub collections (easy navigation)
- Object oriented like experience
- No need for SQL experience
- No need for ORMs and its quirks, ( ORM needs knowledge and experience)
- Observable objects(entities) and tables (collections) wherever changes came from.
- Ultra fast readings.
- Rock solid database.
- LINQ implemented querying.
- "Zero" copy architecture, which leads to minimal RAM consumption.
- You pay (in ressources) only for what you actually consume. Consequence of the previous point .
- Multiple Realms in different locations (mobile or desktop) can be automatically synchronized (MongoDB subscription). 
- You pay only for the synchronized data, not for the offline/local reading/writing.


CONS:

- NEEDS some time for initial adjustments
- Threading model is not trivial: Consequence of the previous point. If you want access your realm from another thread, you should create another realm for the same database on the same new thread then operate on it from there. The main thread's Realm will automatically be synchronized at the end, eventually needs only a Refresh method (I've never needed it)
- Every property change should occure in a transaction, which sometimes can be awkward, especially with UI components. But there are workarounds for this. Feel free to ask your questions here.

---------------------------------

As for your question you can ship your database with your application just like SQLite.

If there's a database file at the RealmConfig specified path, it will be loaded in the Realm. If not, a new empty one will created with a default schema costructed by the object model you define (objects implementing IRealmObject)

**Reply by usefulBeeing** [reply]
Your response was an excellent starting point @uixiu. I found the exact answer to my question while exploring the mongodb documentations under [Bundle a Realm](https://www.mongodb.com/docs/atlas/device-sdks/sdk/dotnet/realm-files/bundle-a-realm/). Thank you for your efforts.

**Reply by uixiu** [reply]
Feel free to ask here if you have  questions.

**Comment by uixiu** [comment]
Hi @usefulBeeing 
I wanted to let you know that I came across a recent MongoDB announcement
https://www.mongodb.com/docs/atlas/device-sdks/deprecation/#std-label-device-sdks-deprecation

The announcement mentions the deprecation of device SDKs (Realm Sync), which could potentially impact Realm in the future. If you plan to rely on device sync functionality, you might want to start exploring alternative solutions.

**Reply by usefulBeeing** [reply]
Thanks for the heads up @uixiu. I appreciate it.
